neutrino-custom/base.js

```diff
@@ -0,0 +1,205 @@
+'use strict';
+const path = require('path');
+const lintPreset = require('./lint');
+const webpack = require('webpack');
+const reactPreset = require('neutrino-preset-react');
+const HtmlPlugin = require('html-webpack-plugin');
+const Md5HashPlugin = require('webpack-md5-hash');
+
+const CWD = process.cwd();
+const SRC = path.join(CWD, 'src'); // neutrino's default source directory
+const UI = path.join(CWD, 'ui');
+const DIST = path.join(CWD, 'dist');
+const INDEX_TEMPLATE = path.join(UI, 'index.html');
+const ADMIN_TEMPLATE = path.join(UI, 'admin.html');
+const PERF_TEMPLATE = path.join(UI, 'perf.html');
+const LOGVIEWER_TEMPLATE = path.join(UI, 'logviewer.html');
+const FAILUREVIEWER_TEMPLATE = path.join(UI, 'failureviewer.html');
+const USERGUIDE_TEMPLATE = path.join(UI, 'userguide.html');
+
+const HTML_MINIFY_OPTIONS = {
+    useShortDoctype: true,
+    keepClosingSlash: true,
+    collapseWhitespace: true,
+    preserveLineBreaks: true
+};
+
+module.exports = neutrino => {
+    reactPreset(neutrino);
+    lintPreset(neutrino);
+
+    // Change the ouput path from build/ to dist/:
+    neutrino.config.output.path(DIST);
+
+    if (process.env.NODE_ENV !== 'test') {
+        // Include files from node_modules in the separate, more-cacheable vendor chunk:
+        const jsDeps = [
+            'angular', 'angular-cookies', 'angular-local-storage', 'angular-resource',
+            'angular-route', 'angular-sanitize', 'angular-toarrayfilter', 'angular-ui-bootstrap',
+            'angular-ui-router', 'bootstrap/dist/js/bootstrap', 'hawk', 'jquery', 'jquery.scrollto',
+            'js-yaml', 'mousetrap', 'react', 'react-dom', 'taskcluster-client'
+        ];
+        jsDeps.map(dep =>
+            neutrino.config.entry('vendor').add(dep)
+        );
+
+        // Ensure chunkhashes for unchanged chunks don't change -- this allows the bundled vendor
+        // file to keep its hash and stay cached in the user's browser when it hasn't been updated.
+        neutrino.config.output.chunkFilename('[chunkhash].[id].chunk.js');
+        neutrino.config.plugin('hash').use(Md5HashPlugin);
```

Comment:
Reading https://webpack.js.org/guides/caching/#generating-unique-hashes-for-each-file left me slightly confused. The docs first give the impression hashed filenames are supported out of the box, then go on to suggest that the default way of generating the hashes causes unnecessary invalidation in some cases, without being very clear as to what they are.

Either way, this seems like something that Neutrino could/should be handling for us. @eliperelman do you agree? If so, would you mind filing a ticket against it?